sympy/core/basic.py

```diff
@@ -1258,6 +1258,33 @@ def has(self, *patterns):
         """
         return self._has(iterargs, *patterns)
 
+    def has_xfree(self, s):
```

Comment:
Maybe this should be type-hinted as `s: set[Basic]`.

Reply:
I don't follow you. `s` should be a set, not Basic.

Reply:
It should be a `set` of `Basic` which is what `set[Basic]` means in typing syntax.